exam.cls

```diff
@@ -34,6 +10,10 @@
 \newif\ifheaderWithID
 \newcommand{\nbBoxesStudentID}{}
 
+\newif\ifanonymousExamen
+\newcommand{\nbBoxesAnonimityNumber}{}
+
+\DeclareOptionX{anonymous}[4]{\anonymousExamentrue\renewcommand{\nbBoxesAnonimityNumber}{#1}}
```

Comment:
Maybe the number of boxes  should be a parameter (with a default value)?

Reply:
I really don't like the option name, but I couldn't find a better identifier. Maybe using two options is a better choice

Reply:
Maybe a better choice would be to allow adding fields independently of the class options.